api/src/main/java/com/strategyobject/substrateclient/api/pallet/polkadot_xcm/PolkadotXcm.java

```diff
@@ -0,0 +1,177 @@
+public interface PolkadotXcm {
+
+    @Event(index = 0)
+    @Getter
+    @Setter
+    @ScaleReader
+    class Attempted {
+        @Scale(ScaleType.U32.class)
+        private Long outcome; // u32 - type 4 - Outcome
+    }
+
+    @Event(index = 1)
+    @Getter
+    @Setter
+    @ScaleReader
+    class Sent {
+        private MultiLocation multiLocation1;
+
+        private MultiLocation multiLocation2;
+
+        private Xcm xcm; // Empty Class
+    }
+
+    @Event(index = 2)
+    @Getter
+    @Setter
+    @ScaleReader
+    class UnexpectedResponse {
+        private MultiLocation multiLocation;
+
+        @Scale(ScaleType.U64.class)
+        private BigInteger queryId; // u64 - type 8 - QueryId
+    }
+
+    @Event(index = 3)
+    @Getter
+    @Setter
+    @ScaleReader
+    class ResponseReady {
+        @Scale(ScaleType.U64.class)
+        private BigInteger queryId; // u64 - type 8 - QueryId
+
+        private Response response; // Empty Class
+    }
+
+    @Event(index = 4)
+    @Getter
+    @Setter
+    @ScaleReader
+    class Notified {
+        @Scale(ScaleType.U64.class)
+        private BigInteger queryId; // u64 - type 8 - QueryId
+
+        @Scale(ScaleType.U8.class)
+        private Integer value1; // u8 - type 2 - u8
+
+        @Scale(ScaleType.U8.class)
+        private Integer value2; // u8 - type 2 - u8
+    }
+
+    @Event(index = 5)
+    @Getter
+    @Setter
+    @ScaleReader
+    class NotifyOverweight {
+        @Scale(ScaleType.U64.class)
+        private BigInteger queryId; // u64 - type 8 - QueryId
+
+        @Scale(ScaleType.U8.class)
+        private Integer value1; // u8 - type 2 - u8
+
+        @Scale(ScaleType.U8.class)
+        private Integer value2; // u8 - type 2 - u8
+
+        @Scale(ScaleType.U64.class)
+        private BigInteger weight1; // u64 - type 8 - Weight
+
+        @Scale(ScaleType.U64.class)
+        private BigInteger weight2; // u64 - type 8 - Weight
+    }
+
+    @Event(index = 6)
+    @Getter
+    @Setter
+    @ScaleReader
+    class NotifyDispatchError {
+        @Scale(ScaleType.U64.class)
+        private BigInteger queryId; // u64 - type 8 - QueryId
+
+        @Scale(ScaleType.U8.class)
+        private Integer value1; // u8 - type 2 - u8
+
+        @Scale(ScaleType.U8.class)
+        private Integer value2; // u8 - type 2 - u8
+    }
+
+    @Event(index = 7)
+    @Getter
+    @Setter
+    @ScaleReader
+    class NotifyDecodeFailed {
+        @Scale(ScaleType.U64.class)
+        private BigInteger queryId; // u64 - type 8 - QueryId
+
+        @Scale(ScaleType.U8.class)
+        private Integer value1; // u8 - type 2 - u8
+
+        @Scale(ScaleType.U8.class)
+        private Integer value2; // u8 - type 2 - u8
+    }
+
+    @Event(index = 8)
+    @Getter
+    @Setter
+    @ScaleReader
+    class InvalidResponder {
+        private MultiLocation multiLocation;
+        private BigInteger queryId; // u64 - type 8 - QueryId
+        private Optional<MultiLocation> optionalMultiLocation;
```

Comment:
I think you need an annotation for Optionals

`@Scale(ScaleType.Option.class)`